behavior_namespace: "/clickable_world"
behavior_names: ["test_button"]
init_conditions: {"at_table" : False, "obj_in_hand" : False}

test_button:
    preconditions: {}
    postconditions: {}
    perception_srv: "/test_button_perception"
    action_srv: "/test_button_action"

move_floor:
    preconditions: {"at_table" : False}
    postconditions: {}
    perception_srv: "/get_empty_floor"
    action_srv: "/move_empty_floor"

table_approach:
    preconditions: {"at_table" : False}
    postconditions: {"at_table" : True}
    perception_srv: "/get_tables"
    action_srv: "/approach_table"

table_backoff:
    preconditions: {"at_table" : True}
    postconditions: {"at_table" : False}
    perception_srv: "/backoff_button"
    action_srv: "/backoff_table"

grasp_object:
    preconditions: {"at_table" : True, "obj_in_hand" : False}
    postconditions: {"obj_in_hand" : True}
    perception_srv: "/get_tables"
    action_srv: "/grasp_obj_on_table"

place_object:
    preconditions: {"at_table" : True, "obj_in_hand" : True}
    postconditions: {"obj_in_hand" : False}
    perception_srv: "/get_tables"
    action_srv: "/place_obj_on_table"

handoff_object:
    preconditions: {"at_table" : False, "obj_in_hand" : True}
    postconditions: {"obj_in_hand" : False}
    perception_srv: "/detect_faces"
    action_srv: "/handoff_obj_to_person"
    
